activestorage/app/models/active_storage/blob.rb

```diff
@@ -157,6 +161,13 @@ def text?
     content_type.start_with?("text")
   end
 
+  def disposition(disposition_override = nil)
+    ActionDispatch::Http::ContentDisposition.format(
+      disposition: disposition_override || "inline",
+      filename: filename.sanitized
+    )
+  end
```

Comment:
This doesn't feel like it quite belongs here, given that it's specific to the proxy controller, and not something intrinsic to blobs. I'd pull it into the controller instead. The object you're using is also a hint that this is a little wonky.